core/storage/filestore/impl/filesystem/filesystem_file.cpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "storage/filestore/impl/filesystem/filesystem_file.hpp"
+
+#include "boost/filesystem.hpp"
+#include "storage/filestore/filestore_error.hpp"
+
+using fc::storage::filestore::FileStoreError;
+using fc::storage::filestore::FileSystemFile;
+using fc::storage::filestore::Path;
+
+FileSystemFile::FileSystemFile(Path path) : path_(std::move(path)) {}
+
+Path FileSystemFile::path() const noexcept {
+  return path_;
+}
+
+fc::outcome::result<size_t> FileSystemFile::size() const noexcept {
+  OUTCOME_TRY(file_exists, exists());
+  if (!file_exists) return FileStoreError::FILE_NOT_FOUND;
+
+  return boost::filesystem::file_size(path_);
+}
+
+fc::outcome::result<void> FileSystemFile::open() noexcept {
+  OUTCOME_TRY(file_exists, exists());
+  if (!file_exists) return FileStoreError::FILE_NOT_FOUND;
```

Comment:
according to style guide need to enclose return with {}
```suggestion
  if (!file_exists) {
    return FileStoreError::FILE_NOT_FOUND;
  }
```

Reply:
Short conditional statements may be written on one line if this enhances readability
In general, curly braces are not required for single-line statements
https://google.github.io/styleguide/cppguide.html#Conditionals